zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/SearchQueryRequestMapper.java

```diff
@@ -147,18 +128,41 @@ private static ProcessInstanceFilter toProcessInstanceFilter(
     final var builder = FilterBuilders.processInstance();
 
     if (filter != null) {
-      final var variableFilters = toVariableValueFilter(filter.getVariables());
-      if (variableFilters != null) {
-        builder.variable(variableFilters);
-      }
-      if (filter.getKey() != null && !filter.getKey().isEmpty()) {
-        builder.processInstanceKeys(filter.getKey());
-      }
+      ofNullable(filter.getRunning()).ifPresentOrElse(builder::running, builder::running);
```

Comment:
This has to do with the other review comment (FilterBuilder). Can this be expressed in more clearer way?